bin/reth/src/args/rpc_server_args.rs

```diff
@@ -477,14 +477,21 @@ impl TypedValueParser for RpcModuleSelectionValueParser {
     ) -> Result<Self::Value, clap::Error> {
         let val =
             value.to_str().ok_or_else(|| clap::Error::new(clap::error::ErrorKind::InvalidUtf8))?;
-        val.parse::<RpcModuleSelection>().map_err(|err| {
-            let arg = arg.map(|a| a.to_string()).unwrap_or_else(|| "...".to_owned());
-            let possible_values = RethRpcModule::all_variants().to_vec().join(",");
-            let msg = format!(
+        val.split(',')
+            .map(str::trim)
+            .collect::<std::collections::HashSet<_>>()
+            .into_iter()
+            .collect::<Vec<_>>()
+            .join(", ")
+            .parse::<RpcModuleSelection>()
+            .map_err(|err| {
+                let arg = arg.map(|a| a.to_string()).unwrap_or_else(|| "...".to_owned());
+                let possible_values = RethRpcModule::all_variants().to_vec().join(",");
+                let msg = format!(
```

Comment:
hmm, it seems like splitting by `,` and trimming is already done in the `FromStr` impl of `RpcModuleSelection` - maybe we should just dedupe in there so this part of the code could just be `RpcModuleSelection::from_str(val)`?

ref https://github.com/paradigmxyz/reth/blob/d890053084e56872ed6415be7fa64e20c432d8be/crates/rpc/rpc-builder/src/lib.rs#L614C18-L625

Reply:
@ungaro pinging in case you missed it